docs/rfcs/XXX-pull-request-review.md

```diff
@@ -0,0 +1,145 @@
+
+When opening a TextEditor on a file that has been annotated with review comments on the current pull request, a block decoration is used to show the comment content at the corresponding position within the file content. Also, a gutter decoration is used to reveal lines that are included within the current pull requests' diff and may therefore include comments.
+
+![in-editor](https://user-images.githubusercontent.com/378023/44790482-69bcc800-abda-11e8-8a0f-922c0942b8c6.png)
+
+> TODO: add gutter decoration?
+
+* The comment's position is calculated from the position acquired by the GitHub API response, modified based on the git diff of that file (following renames) between the owning review's attached commit and the current state of the working copy (including any local modifications). Once created, the associated marker will also track unsaved modifications to the file in real time.
+* The up and down arrow buttons navigate to the next and previous review comments within this review within their respective TextEditors.
+* The "diff" button navigates to the "Changes" tab of the corresponding pull request's `IssueishPaneItem`, expands the owning review, and scrolls to center the same comment within that view.
+
+Hovering along the gutter within a pull request diff region reveals a `+` icon, which may be clicked to begin a new review:
+
+![plus-icon](https://user-images.githubusercontent.com/378023/40348708-6698b2ea-5ddf-11e8-8eaa-9d95bc483fb1.png)
+
+Clicking the `+` reveals a new comment box, which may be used to submit a single comment or begin a multi-comment review:
+
+![single-review](https://user-images.githubusercontent.com/378023/40351475-78a527c2-5de7-11e8-8006-72d859514ecc.png)
+
+* If a draft review is already in progress, the "Add single comment" button is disabled and the "Start a review" button reads "Add review comment".
+* Clicking "Add single comment" submits a non-review diff comment and does not create a draft review.
+* Clicking "Start a review" creates a draft review and attaches the authored comment to it.
+
+## Drawbacks
+
+This adds a substantial amount of complexity to the UI, which is only justified for users that use GitHub pull request reviews.
+
+Showing all reviews in the current pull request tile can easily overwhelm the other pull request information included there. It also limits our ability to expand the information we provide there in the future (like associated issues, say).
+
+Rendering pull request comments within TextEditors can be intrusive: if there are many, or if your reviewers are particularly verbose, they could easily crowd out the code that you're trying to write and obscure your context.
+
+## Rationale and alternatives
+
+<!-- Ongoing --->
+
+## Unresolved questions
+
+### Questions I expect to address before this is merged
+
+Can we access "draft" reviews from the GitHub API, to unify them between Atom and GitHub?
+
+* _Yes, the `reviews` object includes it in a `PENDING` state._
```

Comment:
Do we already know how we're handling starting a review and making sure folks know it hasn't been pushed up to dotcom yet?